vortex-array/src/array/sparse/mod.rs

```diff
@@ -137,11 +137,18 @@ impl SparseArray {
         })
     }
 
-    pub fn min_index(&self) -> usize {
-        let min_index: usize = scalar_at(&self.indices(), 0)
-            .and_then(|s| s.as_ref().try_into())
-            .vortex_expect("Failed to get min_index from SparseArray");
-        min_index - self.indices_offset()
+    /// Return the minimum index if indices are present.
+    ///
+    /// If this sparse array has no indices (i.e. all elements are equal to fill_value)
+    /// then it returns None.
+    pub fn min_index(&self) -> Option<usize> {
+        (!self.indices().is_empty()).then(|| {
+            let min_index: usize = scalar_at(&self.indices(), 0)
+                .and_then(|s| s.as_ref().try_into())
+                .vortex_expect("SparseArray indices is non-empty");
+
+            min_index - self.indices_offset()
```

Comment:
not sure if isempty check is better or just treating the error here as empty?

Reply:
I'm on the fence about it tbh. can change if you think it'd be clearer

Reply:
Ok, we can keep it like this. It will have to change either way